### PR TITLE
Release notes for ipsec-client 3.3.0

### PR DESCRIPTION
**Pull request by agrawalkaushik**

### CODE REVIEW

**Review by MichaelBaj — APPROVED** [review]

File: docs/plugin_ipsec_client.md [hunk 1]
```diff
@@ -52,6 +52,46 @@ exit
...
+exit
+```
+
+| Config   | Description                          |
+| -------- | ------------------------------------ |
+| name     | The name of the libreswan option     |
+| value    | The value of the option              |
+
+:::warning
+The `custom-option` are added to the libreswan config file and any invalid option could prevent the application from starting up.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:92]:
```suggestion
The `custom-option` is added to the libreswan config file; any invalid option could prevent the application from starting up.
```

File: docs/plugin_ipsec_client.md [hunk 2]
```diff
@@ -114,12 +154,14 @@ node
...
-* `enabled` - Allows you to switch tunnel monitoring on and off for a `remote`.
-* `address` - The IP or hostname where traffic is sent. This address must be reachable after traversing the tunnel.
-* `timeout` - Duration (in seconds) within which to reach the destination. Each attempt will be made in this duration / `max-retries` interval.
-* `max-retries` - Number of consecutive missed ICMP ping responses from the destination within the interval before deciding that the tunnel is unhealthy.
-* `interval` - Duration (in seconds) of how often to perform an ICMP probe test to the probe-address.
-* `tunnel-monitor-nat-network` - The subnet where traffic originates. The corresponding ingress KNI's fourth octet is used. By default, the subnet `10.128.128.0/28` is used.
+| Config   | Description                          |
+| -------- | ------------------------------------ |
+| `enabled` | Allows you to switch tunnel monitoring on and off for a `remote`. |
+| `address` | The IP or hostname where traffic is sent. This address must be reachable after traversing the  |tunnel.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:160]:
```suggestion
| `address` | The IP or hostname where traffic is sent. This address must be reachable after traversing the  tunnel. |
```

File: docs/plugin_ipsec_client.md [hunk 3]
```diff
@@ -114,12 +154,14 @@ node
...
+| `timeout` | Duration (in seconds) within which to reach the destination. Each attempt will be made in this  |duration / `max-retries` interval.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:161]:
```suggestion
| `timeout` | Duration (in seconds) within which to reach the destination. Each attempt will be made in this duration / `max-retries` interval. |
```

File: docs/plugin_ipsec_client.md [hunk 4]
```diff
@@ -114,12 +154,14 @@ node
...
+| `max-retries` | Number of consecutive missed ICMP ping responses from the destination within the interval  |before deciding that the tunnel is unhealthy.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:162]:
```suggestion
| `max-retries` | Number of consecutive missed ICMP ping responses from the destination within the interval before deciding that the tunnel is unhealthy. |
```

File: docs/plugin_ipsec_client.md [hunk 5]
```diff
@@ -114,12 +154,14 @@ node
...
+| `interval` | Duration (in seconds) of how often to perform an ICMP probe test to the probe-address. |
+| `tunnel-monitor-nat-network` | The subnet where traffic originates. The corresponding ingress KNI's fourth  |octet is used. By default, the subnet `10.128.128.0/28` is used.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:164]:
```suggestion
| `tunnel-monitor-nat-network` | The subnet where traffic originates. The corresponding ingress KNI's fourth octet is used. By default, the subnet `10.128.128.0/28` is used. |
```

File: docs/plugin_ipsec_client.md [hunk 6]
```diff
@@ -469,6 +533,47 @@ Completed in 0.10 seconds
...
+
+- **PLUGIN-1554** Collect ipsec-client plugin data as part of tech support info
+
+The detailed tech support info bundle will now include the necessary logs and data for troubleshooting ipsec-client plugin related problems.
+
+- **PLUGIN-1598** - Support custom options for obscure libreswan config pieces
+
+A new config option has been added to enable libreswan features that are not made available as first class configuration.
+
+- **PLUGIN-1591** Add cpu, memory & status tracking for ipsec services
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:563]:
```suggestion
- **PLUGIN-1591** Add cpu, memory, and  status tracking for ipsec services.
```

File: docs/plugin_ipsec_client.md [hunk 7]
```diff
@@ -469,6 +533,47 @@ Completed in 0.10 seconds
...
+
+The system will now track the cpu, memory and usage information for various IPSec client related processes.
+
+#### Issues Fixed
+
+- **PLUGIN-1628** Incorrect network-script path was being used in the auto generated device-interface configuration
+
+  _**Resolution**_ The auto configuration was updated to use the right script path.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/plugin_ipsec_client.md:571]:
```suggestion
  _**Resolution**_ The auto configuration was updated to use the correct script path.
```

**Review by Chr1st0ph3rTurn3r — CHANGES_REQUESTED** [review]

File: docs/plugin_ipsec_client.md [hunk 8]
```diff
@@ -469,6 +533,47 @@ Completed in 0.10 seconds
...
+
+- **PLUGIN-1532** Enable additional configuration options for IPSec tunnels.
+
+The `encapsulation` and `remote-peer-type` options are now available for configuration.
+
+- **PLUGIN-1554** Collect ipsec-client plugin data as part of tech support info
+
+The detailed tech support info bundle will now include the necessary logs and data for troubleshooting ipsec-client plugin related problems.
+
+- **PLUGIN-1598** - Support custom options for obscure libreswan config pieces
```
Comment (BenMatase) [inline docs/plugin_ipsec_client.md:559]: `pieces` -> `fields`?

File: docs/plugin_ipsec_client.md [hunk 9]
```diff
@@ -52,6 +52,46 @@ exit
 This plugin can only connect to IPsec endpoints that support pre-shared key authentication.
 :::
 
+#### Custom Options
+
+##### Version History
+
+| Release  | Modification                         |
+| -------- | ------------------------------------ |
+| 3.3.0    | `profile > custom-option` introduced |
```
Comment (BenMatase) [inline docs/plugin_ipsec_client.md:61]: Add the version history to new command?

**Review by BenMatase — CHANGES_REQUESTED** [review]

File: docs/plugin_ipsec_client.md [hunk 10]
```diff
@@ -444,6 +486,28 @@ If tunnel monitoring is enabled for a remote, corresponding tunnel monitoring st
            Last Restart:2022-02-14 15:41:15
 ```
 
+### Commands
+The `restart ipsec tunnel` command can be used to restart an individual IPSec tunnel via the conductor PCLI and UI.
```
[the remark this answers is not in the document]
Comment (agrawalkaushik) [reply docs/plugin_ipsec_client.md:490]:
```suggestion
The `restart ipsec remote` command can be used to restart an individual IPSec tunnel via the conductor PCLI and UI.
```

**Review by Chr1st0ph3rTurn3r — DISMISSED** [review]

**Review by Chr1st0ph3rTurn3r — APPROVED** [review]

**Review by Chr1st0ph3rTurn3r — APPROVED** [review]

**Pull request merged**